File: typescript/structural/bridge/remote_device/tv.ts
import {Device} from "./device";

export class Tv implements Device {
  able: boolean
  volume: number
  channel: number

  constructor() {
    this.able = Math.random() < 0.5 ? true : false
    this.volume = Math.round(Math.random() * 100)
    this.channel = Math.round(Math.random() * 100)
  }


  disable(): boolean {
    return false;
  }

  enable(): boolean {
    return true;
  }

  getChannel(): number {
    return this.channel;
  }

  getVolume(): number {
    return this.volume;
  }

  isEnabled(): boolean {
    return this.able;
  }

  setChannel(channel: number): number {
    this.channel = channel
    return this.channel;
  }

  setVolume(percent: number): number {
    this.volume = percent
    return this.volume;
  }

}

File: dart/structural/bridge/remote_device/bin/tv.dart
import 'dart:math';

import './device.dart';

class Tv implements Device {
  bool able;
  int volume;
  int channel;

  Tv() {
    Random rand = Random();
    able = rand.nextBool();
    volume = rand.nextInt(100);
    channel = rand.nextInt(100);
  }

  @override
  bool disable() {
    able = false;
    return able;
  }

  @override
  bool enable() {
    able = true;
    return true;
  }

  @override
  int getChannel() {
    return channel;
  }

  @override
  int getVolume() {
    return volume;
  }

  @override
  bool isEnabled() {
    return able;
  }

  @override
  int setChannel(int channelValue) {
    channel = channelValue;
    return channel;
  }

  @override
  int setVolume(int volumeValue) {
    volume = volumeValue;
    return volume;
  }
}
